### PR TITLE
Migrate componenttest factories to new the interface

### PR DESCRIPTION
**Pull request by dmitryax**
**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/1542

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -10,6 +10,7 @@ cloud.google.com/go v0.50.0/go.mod h1:r9sluTvynVuxRIOHXQEHMFffphuXHOMZMycpNR5e6T
 cloud.google.com/go v0.51.0/go.mod h1:hWtGJ6gnXH+KgDv+V0zFGDvpi07n3z8ZNj3T1RW0Gcw=
 cloud.google.com/go v0.52.0/go.mod h1:pXajvRH/6o3+F9jDHZWQ5PbGhn+o8w9qiu/CffaVdO4=
 cloud.google.com/go v0.53.0/go.mod h1:fp/UouUEsRkN6ryDKNW/Upv/JBKnv6WDthjR6+vze6M=
+cloud.google.com/go v0.56.0 h1:WRz29PgAsVEyPSDHyk+0fpEkwEFyfhHn+JbksT6gIL4=
```
Comment (bogdandrutu) [inline go.sum:13]: Why this change?

File: service/builder/pipelines_builder_test.go [hunk 2]
```diff
@@ -203,86 +197,32 @@ func TestPipelinesBuilder_BuildVarious(t *testing.T) {
...
-					Nanos:   456,
-				},
-				EndTime: &timestamp.Timestamp{
-					Seconds: 123456789,
-					Nanos:   456,
-				},
-			},
-		},
-	}
+func generateTestTraces() pdata.Traces {
```
Comment (bogdandrutu) [inline service/builder/pipelines_builder_test.go:200]: Use testdata package for this

File: go.sum (hunk 1 above)
Comment (dmitryax) [reply go.sum]: `make test` :) . I removed it

File: service/builder/pipelines_builder_test.go (hunk 2 above)
Comment (dmitryax) [reply service/builder/pipelines_builder_test.go]: addressed
